setup.py

```diff
@@ -492,7 +493,94 @@ def get_submodule_folders() -> list[Path]:
         ]
 
 
-def check_submodules() -> None:
+def initialize_git_repository() -> None:
+    """Initialize the git repository if it does not already exist."""
+    if (CWD / ".git").exists():
+        # If the .git directory exists, we assume the repository is already
+        # initialized via a git clone.
+        return
+
+    report(" --- Initializing git repository")
+    start = time.perf_counter()
+
+    commands = (
+        ["git", "init", "--initial-branch=main"],
```

Comment:
I don't like initializing a new branch. I wish we could do add just the depth1 tree into this folder somehow?

Reply:
Also, the overhead of creating a fresh git repository is very small. Because we already have the necessary files in the build folder.

Reply:
> Why is the `.git` file even needed?

This is used to make `git submodule update --init --recursive` work.

We do not need to rely on the superproject's `.git`. Instead, we can use `git clone --recurse-submodules` the third-party repos one by one.